tests/net/lib/dns_cache/src/main.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024 Endress+Hauser AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/fff.h>
+#include <zephyr/ztest.h>
+#include <zephyr/logging/log.h>
+#include "dns_cache.h"
+LOG_MODULE_REGISTER(net_dns_cache_test, CONFIG_DNS_RESOLVER_LOG_LEVEL);
+
+#define TEST_DNS_CACHE_SIZE        12
+#define TEST_DNS_CACHE_DEFAULT_TTL 1
+DNS_CACHE_DEFINE(test_dns_cache, TEST_DNS_CACHE_SIZE);
+
+void clear_cache(void)
+{
+	dns_cache_flush(&test_dns_cache);
+}
+
+ZTEST_SUITE(net_dns_cache_test, NULL, NULL, clear_cache, NULL, NULL);
+
+ZTEST(net_dns_cache_test, test_simple_cache_entry)
+{
+	struct dns_addrinfo info_write = {.ai_family = 42};
```

Comment:
Is the 42 value here just some arbitrary value, why not use proper address family values?

Reply:
It is not arbitrary (https://en.wikipedia.org/wiki/Phrases_from_The_Hitchhiker%27s_Guide_to_the_Galaxy#Answer_to_the_Ultimate_Question_of_Life,_the_Universe,_and_Everything_(42)) but definitely only a sentinel value 😅.

I have replaced it with a proper address family.